refactor: added utility and spacing classes and refactored

TejasShekar opened the pull request.

TejasShekar committed:
refactor: added utility and spacing classes and refactored all components

TejasShekar committed:
fix: resolved all merge conflicts

TejasShekar committed:
styles: changed primary-dark color

This provides better contrast ratio

TejasShekar committed:
styles: fix aside position issue

Maianki commented on Components/Buttons/button.css:
```diff
@@ -1,27 +1,33 @@
-  font-size: 1.5rem;
+  font-size: 1.3rem;
   border: 1px solid transparent;
   border-radius: 0.475rem;
```
Any specific reason for using such a specific value?

TejasShekar replied:
default font size of button text is too small to read. I felt 1.3rem is a sweet spot and has good readbility. So, hardcoded the value.

ravikumar0403 commented on Stylesheets/reset.css:
```diff
@@ -0,0 +1,69 @@
+  text-align: center;
+  outline: none;
+  border: none;
+  /* background-color: white; */
```
remove this commented code

TejasShekar replied:
done

ravikumar0403 reviewed (approved):
Looks good to me

Maianki commented on Components/Input/input.css:
```diff
@@ -21,19 +20,15 @@ input[type="password"] {
 #success-input {
-  border: 3px solid var(--success);
+  border: 2px solid var(--success);
 }
 
 #error-input {
-  border: 3px solid var(--danger);
-}
-
-.gap-0 {
-  gap: 0;
+  border: 2px solid var(--danger);
```
Suggestion: Avoid using Id selectors until it's really necessary as they have very high specificity.

TejasShekar replied:
Will add a proper class and fix it.

MaruthiPothuganti commented on Components/Input/input.css:
```diff
@@ -1,6 +1,5 @@
 .input-container {
   width: 300px;
```
This will hinder responsiveness. You can use REMs instead

darshandamre reviewed (approved):
good job Tejas

Maianki reviewed (approved):
Looks good just a small suggestion try raising small PR .

MaruthiPothuganti commented on Stylesheets/docs.css:
```diff
@@ -0,0 +1,64 @@
+  color: var(--primary-dark);
+}
+
+span {
```
Avoid styling with tag names. Use a class for span and apply  styling to that.

MaruthiPothuganti commented on Stylesheets/docs.css:
```diff
@@ -0,0 +1,64 @@
+  font-size: 1.1rem;
+}
+
+aside {
```
Same as above. Avoid styling with tag names

MaruthiPothuganti commented on Stylesheets/reset.css:
```diff
@@ -0,0 +1,69 @@
+button {
+  cursor: pointer;
+  text-align: center;
+  outline: none;
```
Its not a good practice to make Outline of a button as none as it. It provides visual feedback for links that have "focus" when navigating a web document using the TAB key (or equivalent). This is especially useful for folks who can't use a mouse or have a visual impairment. If you remove the outline you are making your site inaccessible for these people

MaruthiPothuganti reviewed (commented):
Suggested a few changes. Everything else looks good. The code is well organised.

TejasShekar committed:
fix: PR suggestions

1. Added class name to elements that was being styles using their tag name.
2. Changed input-container width from px to rem for responsiveness
3. removed outline and bg-color property of button element in reset.css

TejasShekar committed:
refactor: selectors in input component

replaced id selectors with class selectors

TejasShekar commented:
Implemented all PR suggestions

TejasShekar committed:
minor refactor

1. added missing span class
2. updated utility classes

TejasShekar committed:
minor refactor

1. added gap class
2. fixed span text color
